toctoc/core/src/main/scala/io.buildo.toctoc/core/authentication/TokenBasedRecovery.scala

```diff
@@ -0,0 +1,42 @@
+package io.buildo.toctoc
+package core
+package authentication
+
+import cats.Monad
+import cats.data.EitherT
+
+import java.time.{Duration, Instant}
+
+object TokenBasedRecovery {
+  import TokenBasedAuthentication._
+
+  class TokenBasedRecoveryFlow[F[_]: Monad](
+    loginD: LoginAuthenticationDomain[F],
+    recoveryTokenD: AccessTokenAuthenticationDomain[F],
+    tokenDuration: Duration = Duration.ofDays(365),
+  ) extends BCryptHashing {
+    def registerForRecovery(s: Subject): F[Either[AuthenticationError, AccessToken]] =
+      (for {
+        rtd <- EitherT(recoveryTokenD.unregister(s))
+        token = AccessToken(randomString(64), Instant.now().plus(tokenDuration))
```

Comment:
maybe we can add a constructor to `AccessToken` that uses these defaults:

```scala
object AccessToken {
  def generate(duration: Duration): AccessToken = AccessToken(randomString(64), Instant.now.plus(duration)
}
```

and then

```suggestion
        token = AccessToken.generate(tokenDuration)
```

Reply:
Nice idea, I'd leave `randomString` out of scope and open new issues for:
- `randomString` comes from `BCryptHashing`
- `Instant.now` could be handled in a more generic way, perhaps something like `Clock[F[_]]` from cats-effect.

Reply:
Sounds good. Let's open new issues

Reply:
done ✅